subsys/bluetooth/controller/ll_sw/openisa/lll/pdu_vendor.h

```diff
@@ -4,33 +4,27 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/sys/util.h>
+
 /* Minimum vendor specific Rx payload buffer allocation */
 #define LL_VND_OCTETS_RX_MIN 0
 
 /* No vendor Data PDU struct octet3 */
 struct pdu_data_vnd_octet3 {
-	union {
-		uint8_t resv[0];
-	} __packed;
+	FLEXIBLE_ARRAY_DECLARE(uint8_t, resv);
```

Comment:
I have concern (not verified) that `__packed` has been removed by the introduction of `FLEXIBLE_ARRAY_DECLARE`. Is this truly equivalent?

I see prior PRs have already change some in `subsys/bluetooth/controller/ll_sw/pdu.h` file.

Can we have a `__packed` variant of this "workaround" define, like `FLEXIBLE_ARRAY_DECLARE_PACKED`?

Reply:
And even if it wasn't of size 0, since the elements are byte aligned, I don't think `__packed` would do anything at all.

Reply:
I think this example is wrong. With/without `packed` you will get 4 bytes if `CONFIG_BT_CTLR_DATA_LENGTH_CLEAR` is not defined. And since this union is the only element of the structure `pdu_data_vnd_octet3` that is packed you will get 1 byte size when `CONFIG_BT_CTLR_DATA_LENGTH_CLEAR` is defined.

If you noticed, I have just changed for this specific use case where the union.

I am wondering if I am missing something here though. Did you get any memory alignment issue ? Printing the struct sizes I get exactly the same thing and the alignment should be correct.